src/hb-ot-color-colr-table.hh

```diff
@@ -120,6 +155,89 @@ struct COLR
 			  (this+layersZ).sanitize (c, numLayers)));
   }
 
+  template<typename Iterator,
+           hb_requires (hb_is_iterator (Iterator))>
+  bool serialize (hb_subset_context_t *c,
```

Comment:
We prefer to have serialize not take a subset_context where possible (so that serialize can be used outside of subsetting). Since this table is pretty straightforward that should be possible. You could change this function to take a serialize context, the version and two iterators, one that iterates through BaseGlyphRecord's and a second that iterates through the LayerRecord's. The iterators should yield the final record (with gid's and layer indices already remapped) at each iteration instead of relying on the serialize function to fix those. The remapping logic would then be moved up into the subset call.

Reply:
That makes sense. Thanks for the insight/suggestion.